### PR TITLE
replace instances of PACKAGE_VERSION with Get function

### PR DESCRIPTION
**Pull request by andy5995**
The #ifndef can probably be removed after the scons build is deprecated

### CODE REVIEW

**Review by winrid — APPROVED** [review]

**Comment by andy5995** [comment]
@winrid This will need to be re-reviewed. I made several changes after you approved.

File: src/Lib/globals.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+#include "globals.hpp"
+
+#ifndef PACKAGE_VERSION
+#include "config.h"
+#endif
+
+std::string GetVersion(void)
```
Comment (winrid) [inline src/Lib/globals.cpp:8]: Actually I think having a global GetVersion method is kinda ambiguous, there are many things that can be versioned, like network protocols etc. Maybe change the name or add a class and make this a static method like Package::GetVersion?

Comment (andy5995) [reply src/Lib/globals.cpp]: > Actually I think having a global GetVersion method is kinda ambiguous, there are many things that can be versioned, like network protocols etc. Maybe change the name or add a class and make this a static method like Package::GetVersion?

Sounds reasonable. Done.

File: src/Lib/globals.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#include "globals.hpp"
+
+#ifndef PACKAGE_VERSION
+#include "config.h"
+#endif
+
+const std::string Package::GetVersion(void)
```
Comment (winrid) [inline src/Lib/globals.cpp:25]: sorry - can we rename this to package.cpp/hpp now? worried people will try to treat globals like "utils" :P

Comment (andy5995) [reply src/Lib/globals.cpp]: > sorry - can we rename this to package.cpp/hpp now? worried people will try to treat globals like "utils" :P

No apology necessary. Done.

Let's remember to "squash and merge" this PR when merging.

**Comment by winrid** [comment]
did squash merge this time :) I could have sworn github had a way to set that as the default but I guess not

**Comment by andy5995** [comment]
No default that I know of. It seems that GitHub normally makes the "default" the last one used.

**Pull request merged**